core/common/src/main/java/alluxio/underfs/ObjectUnderFileSystem.java

```diff
@@ -428,6 +538,32 @@ protected String convertToFolderName(String key) {
   protected abstract boolean deleteObject(String key) throws IOException;
 
   /**
+   * Internal function to delete a list of keys.
+   *
+   * @param keys the list of keys to delete
+   * @return list of successfully deleted keys
+   */
+  protected List<String> deleteObjects(List<String> keys) throws IOException {
+    List<String> result = new LinkedList<>();
+    for (String key : keys) {
```

Comment:
This will only parallelize deletes in batches of the listing size? If the listing size is 1000, then deleting 1000 files will still take O(1000), right?